### DISCUSSION TITLE
Bad type for argument 1 of SetLong ( string vs function )

### DISCUSSION THREAD

**Discussion by torinwalker** [fluffos/fluffos#1093] [Q&A]
I have DeadSoulds 3.9 running with a very old bundled version of Fluffos-2.23-ds03. Am now trying to integrate with the latest fluffos-20230904-3279863 release. 

I have configured fluffos so far to be compatible with the older code such as statics, array keywords, etc., and I am able to run the mudlib and interact with it, but througout the mudlib code, the use of  (: functions :) is throwing Bad argument errors. The same code works in fluffos-2.23-ds03.

```
Bad type for argument 1 of SetLong ( string vs function )
  SetLong( (: LongString :) )
```
Could there be a #define that I am missing? I have searched throughout the code, including make configs, local_options, and even deeper in the source, but cannot find anything related to how (: functions :) are handled by the compiler. Even the present-day LPC references (and within the Fluffos distro source itself!) suggest that (: someFunc :) is valid code. So why is this happening?

Here is some code that reproduces the error in my environment (ds3.9 has hundreds of instances of functions as arguments):

```
#include <lib.h>
#include ROOMS_H

inherit LIB_ROOM;

string descFunc() {
  return "Description: This function-type description fails with bad argument 1";
}

static void create() {
    room::create();

    SetClimate("indoors");
    SetAmbientLight(40);
    SetShort("test room");
    SetLong( (: descFunc :) );  // <---- this fails (gives "Bad type for argument 1" error)
    //SetLong("This works just fine as a string." ); // <---- this works fine, as one would expect
}

void init(){
    ::init();
}

```

**Comment by michaelprograms** [comment]
I think earlier versions of FluffOS were less strict about mismatched variable types. This is more of a mudlib issue than a FluffOS issue. As a note, FluffOS will pass functions around as either `function` or `mixed` type, but what's happening here is its trying to store a `function` in a `string`.

So the `SetLong` in the room inheritable for Dead Souls looks fine, it takes a mixed and passes along to `SetInternalDesc`: https://github.com/fluffos/dead-souls/blob/09a74caa87d8aadbfe303c294cc0bebb25fdb4db/lib/lib/std/room.c#L816-L818

But in the file that defines `SetInternalDesc`, it defines `private string InternalDesc` instead of `mixed`: https://github.com/fluffos/dead-souls/blob/09a74caa87d8aadbfe303c294cc0bebb25fdb4db/lib/lib/events/look_in.c#L11

`GetInternalDesc()`  is checking if an array, function, or string is in this variable, so it should be defined as `private mixed InternalDesc = 0;`.
